Yearn-agents/InstantWithdrawl/src/agent.ts

```diff
@@ -0,0 +1,59 @@
+import { BlockEvent, Finding, HandleBlock, getJsonRpcUrl } from "forta-agent";
+import axios, { AxiosPromise } from "axios";
+import { generateFinding, getAccounts, Mapping } from "./utils";
+import Web3 from "web3";
+import runServer from "./blockchain";
+import { withdrawAbi } from "./abi";
+
+const providerHandleBlock = (Web3: any, axios: any): HandleBlock => {
+  return async (blockEvent: BlockEvent) => {
+    const findings: Finding[] = [];
+
+    const { mapping, users } = await getAccounts(axios);
+
+    const vaults = Object.keys(mapping);
+
+    const server = await runServer(Array.from(users) as string[]);
```

Comment:
I am also wondering if this will work as expected. You are using the `port: 9545` and the agent can handle many block asynchronously, then if all the agents are trying to mount a server at the same port, I guess it will fail.

Reply:
good point. But that is not something that can be resolved by us. The best way for this would be for forta themselves provide a way to get a unique free port everytime. I already mentioned that in the feedback call.

Reply:
Actually, you can use `ganache.provider` instead of `ganache.server` for forking, in that way you don't need a node.

Reply:
i just checked this. Yes what you are saying can reduce the lines of code a bit. But if you see, if we don't spin a server ourselves, it is done internally. So sticking to this approach is better since it is much more clear in this way that a fork server is initiated.

Reply:
Also, just using the provider wont make it work any way. Provider is an object we use in order to communicate to a particular rpc endpoint. So if you wanna stimulate, ie. fork, a server is mandatory. Using ganache.provider directly internally starts one that all.